api/api.go

```diff
@@ -265,3 +275,31 @@ func NewServerRoutes(s *Server) *http.ServeMux {
 
 	return m
 }
+
+// makeDagReceiver constructs a Receivers (HTTP router) from a qri p2p node
+func makeDagReceiver(node *p2p.QriNode) (*dsync.Receivers, error) {
+	ipfsn, err := node.IPFSNode()
+	if err != nil {
+		return nil, err
+	}
+	ng := &dag.NodeGetter{Dag: coreapi.NewCoreAPI(ipfsn).Dag()}
+
+	capi, err := newIPFSCoreAPI(node)
+	if err != nil {
+		return nil, err
+	}
+	bapi := capi.Block()
+
+	return dsync.NewReceivers(context.Background(), ng, bapi), nil
+}
+
+// newIPFSCoreAPI constructs a CoreAPI (part of ipfs) from a qri p2p node
+// copied from qri/actions/dag.go
```

Comment:
opened #720 to deal with this. Let's get this merged then refactor away the need for this func